File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.java


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

	static class Coordinate implements Comparable<Coordinate> {
		int x;
		int y;

		public Coordinate(int x, int y) {
			super();
			this.x = x;
			this.y = y;
		}

		@Override
		public int compareTo(Coordinate o) {
			if (this.x == o.x) {
				return this.y - o.y;
			}
			return this.x - o.x;
		}

	}

	// Main
	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(br.readLine());
		Coordinate[] arr = new Coordinate[N];
		StringTokenizer st;
		for (int i = 0; i < N; i++) {
			st = new StringTokenizer(br.readLine());
			arr[i] = new Coordinate(Integer.parseInt(st.nextToken()), Integer.parseInt(st.nextToken()));
		}
		Arrays.sort(arr);
		StringBuilder sb = new StringBuilder();
		for(Coordinate ele :arr) {
			sb.append(ele.x).append(' ').append(ele.y).append("\n");
		}
		System.out.println(sb);

	}
}

File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.py
class Coordinate:
    def __init__(self, x,y):
        self.x = x
        self.y =y
    def __lt__(self, other):
        if(self.x ==other.x):
            return self.y < other.y
        return self.x<other.x
arr = []
N = int(input())
for n in range(N):
    x,y = map(int, input().split())
    arr.append(Coordinate(x,y))
arr.sort()
for cool in arr:
    print(cool.x, cool.y)